sentry-core/src/main/java/io/sentry/core/Session.java

```diff
@@ -186,11 +186,13 @@ public void end(final @Nullable Date timestamp) {
       if (timestamp != null) {
         this.timestamp = timestamp;
       } else {
-        this.timestamp = DateUtils.getCurrentDateTime();
+        this.timestamp = DateUtils.getCurrentDateTimeOrNull();
       }
 
-      duration = calculateDurationTime(this.timestamp);
-      sequence = getSequenceTimestamp(this.timestamp);
+      if (this.timestamp != null) {
```

Comment:
Likely worth not having sessions enabled at all for this device since they'll be broken

Reply:
yep, `SessionAdapter` will drop on deserialization at least.